synapse/storage/databases/main/room.py

```diff
@@ -1382,6 +1382,25 @@ async def is_partial_state_room_batched(
         partial_state_rooms = {row[0] for row in rows}
         return {room_id: room_id in partial_state_rooms for room_id in room_ids}
 
+    @cached(max_entries=10000, iterable=True)
+    async def get_partial_rooms_for_user(self, user_id: str) -> AbstractSet[str]:
```

Comment:
If the set of partially stated rooms is small (which we're assuming anyway with this query), could we just have a single cached list of the partially stated rooms.

```suggestion
    @cached()
    async def get_partial_rooms(self) -> AbstractSet[str]:
```

Reply:
Oh, yeah that is a better idea!